### ISSUE TITLE
[feature] Add an ability to build dependencies in parallel

### ISSUE THREAD

**Issue by MichalCermakx** [conan-io/conan#14570]
### What is your suggestion?

Hi Conan community,

I know there is an ability to download packages in parallel via general.parallel_download. I have a use case, where "conan install" builds a large number of dependency packages. In my case build means downloading/decompressing those from my company's share drive containing non-conan builds.
I would like to see a feature for the parallel build of dependency packages during conan install/create stage. This might significantly improve dependency installation time.

Thanks, Michal

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @MichalCermakx 

Thanks for your suggestion.

The reasons why there are not parallel builds at the moment are:
- often, using the machine resources to build (compile from sources) a given package, already employs efficiently the computer resources, both because compilation is heavy, but also because it parellelizes good
- For general build parallelism, Conan approaches the problem as a decentralized one, with Conan computing what needs to be built, and the distributing the load into the different build agents. This is typically done with a ``conan graph build-order`` (with ``build-order-merge`` you can also aggregate builds for multiple products and versions), and then iterating the result, sending the build of that package to an agent.

I think it might be doable to try to parallelize also a single conan command building multiple packages, but I am afraid at the moment this will not be a priority. It is likely that we need to improve first the Conan cache concurrency (at the moment it is not concurrent), and this kind of parallelization is usually not that straightforward, having many other side effects (for example, we would need to change how the whole output to stdout is managed, so the different build commands outputs are not scrambled, and that can be a relevant effort). Adding this for consideration in the 2.X roadmap.

**Comment by memsharded** [comment]
Closing in favor of https://github.com/conan-io/conan/issues/15840 where the Conan 2 cache concurrency future progress will be reported.
